### ISSUE TITLE
Graphite Web Proxy Support (Annotations)

### ISSUE THREAD

**Issue by Shalankwa** [sitespeedio/sitespeed.io#3892]
### Feature/improvement

Currently, there is no way (that I can see) to send annotations to a graphite web instance that is behind a proxy layer; changing the full path to the /events endpoint. Trying to do this by setting the --graphite.webHost flag to include the proxy path causes an error on 'getaddrinfo'. 

`{"errno":-3008,"code":"ENOTFOUND","syscall":"getaddrinfo","hostname":"<host>/<proxy-path>"}
    at start (file:///usr/src/app/bin/sitespeed.js:146:15)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at runNextTicks (node:internal/process/task_queues:64:3)
    at process.processImmediate (node:internal/timers:447:9)
    at async file:///usr/src/app/bin/sitespeed.js:178:1`

If there could be a way to overwrite the default PATH to the graphite [events](https://github.com/sitespeedio/sitespeed.io/blob/main/lib/plugins/graphite/send-annotation.js#L98) endpoint then graphite web instances that are behind a proxy can be better supported.

Suggestion:
adding a config flag like "--graphite.proxyPath" that would be added to the **webHost** url on requests to graphite

**Comment by soulgalore** [comment]
Hi @Shalankwa thanks for creating the issue, can you give me a full example of what it will look like? I guess the path could be /events by default and then just changed if you set the parameter?

**Comment by Shalankwa** [comment]
> Hi @Shalankwa thanks for creating the issue, can you give me a full example of what it will look like? I guess the path could be /events by default and then just changed if you set the parameter?

Yes I think /events as the default, and only different when the param has been set.

currently our graphite web event endpoint sits at a full path like: www.domain.com/api/v1/graphite/events/, so the params we would provide in this example are
--graphite.webHost = www.domain.com
--graphite.proxyPath = /api/v1/graphite/

Within the send-annotation.js it would add the proxyPath to the default /events/. It doesn't look like you use the webHost config param anywhere else, but this way would be best to ensure you don't need to refactor anything for the "getaddrinfo" validation that is performed on the domain.

Alternatively, you could allow adding paths in the webHost config param and parse any added path off of it before validating the domain.

**Comment by Shalankwa** [comment]
I created an example PR with the changes that would allow this to work... I apologize if I missed any contributing rules.

**Comment by Shalankwa** [comment]
While I only submitted this for graphite, this could potentially also apply to grafana if it is also served behind a proxy.

**Comment by Shalankwa** [comment]
Hi @soulgalore ,

Thank you very much for merging the PR. There was one change you made that affect how the PATH got made.
You changed the logic 
from:
path: options.graphite.proxyPath + '/events/'
to:
path: options.graphite.proxyPath || '' + '/events/',

Which actually changes it to only be set to 'proxyPath' or '/events/'. Not the logic in my PR where it sets it as proxyPath + '/events/' with the default (only in CLI - I see this change was to fix this) being an empty proxyPath. 

This behavior is actually fine, as someone might actually want to change the path entirely (or integrate it with their own API) which adding /events/ could affect, but the additional string concatenation of the empty string is irrelevant at that point.

Basically, the original logic was
(<proxyPath>/events/') || ('/events/')

And the changed logic is:
('<proxyPath>') || ('/events/')

Again, this doesn't actually need to be changed, just something I wanted to note as the empty string might cause confusion in debugging.

**Comment by soulgalore** [comment]
Ah thanks, yes that was a mistake by me, let me fix that for the next release!

**Comment by soulgalore** [comment]
This was fixed if I remember it correctly.
